lib/server/services/dpl/getPhysicalDplDetectorOrFail.js

```diff
@@ -20,21 +21,33 @@ const NON_PHYSICAL_DETECTORS = ['TST'];
 /**
  * Find and return a physical detector identified by its name or id and throw if it does not exists
  *
- * @param {number} dplDetectorId the id of the DPL detector to fetch
+ * @param {DplDetectorIdentifier} identifier the identifier of the DPL detector to fetch
  * @return {Promise<DplDetector>} the found DPL detector
  */
-exports.getPhysicalDplDetectorOrFail = async (dplDetectorId) => {
+exports.getPhysicalDplDetectorOrFail = async ({ dplDetectorId, dplDetectorName }) => {
+    let criteria;
+    if (dplDetectorId !== undefined) {
```

Comment:
Quick Q, why are you not also comparing to `null` in a similar fashion you did above:  https://github.com/AliceO2Group/Bookkeeping/pull/1565/files#diff-bd788afff7eb270ed28f9fedd009f502eff3f9ce345403c56acff2eee9b8de91R29

Reply:
Because in the cases you mention I simply moved code. But we don't expect it to be null, hence no need to check it. I will remove the check in the other functions